src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionExecutor.cs

```diff
@@ -52,19 +53,7 @@ public static class ControllerActionExecutor
                     }
                     else
                     {
-                        var defaultValueAttribute = 
-                            parameterInfo.GetCustomAttribute<DefaultValueAttribute>(inherit: false);
-
-                        if (defaultValueAttribute?.Value == null)
-                        {
-                            value = parameterInfo.ParameterType.GetTypeInfo().IsValueType
-                                ? Activator.CreateInstance(parameterInfo.ParameterType)
-                                : null;
-                        }
-                        else
-                        {
-                            value = defaultValueAttribute.Value;
-                        }
+                        value = actionMethodExecutor.GetDefaultValueForParameter(index);
```

Comment:
Deal with `parameterInfo.DefaultValue` inside `GetDefaultValueForParameter()`. No need to expose `ActionParameters` or to split the default value logic into two places.